Lib/tkinter/commondialog.py

```diff
@@ -37,22 +37,17 @@ def show(self, **options):
 
         self._fixoptions()
 
-        # we need a dummy widget to properly process the options
-        # (at least as long as we use Tkinter 1.63)
-        w = Frame(self.master)
-
+        master = self.master
+        if master is None:
+            master = _get_temp_root()
         try:
-
-            s = w.tk.call(self.command, *w._options(self.options))
-
-            s = self._fixresult(w, s)
-
+            self._test_callback(master)
```

Comment:
This puzzled me until I read the tests.  Perhaps add comment
```suggestion
            self._test_callback(master)  # The function below is replaced for some tests.
```